components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/fetch/GeckoViewFetchClient.kt

```diff
@@ -107,11 +109,16 @@ private fun WebRequest.Builder.addBodyFrom(request: Request): WebRequest.Builder
     return this
 }
 
-private fun WebResponse.toResponse(): Response {
+@VisibleForTesting
+internal fun WebResponse.toResponse(): Response {
     val headers = translateHeaders(this)
+    val isBlobUrl = uri.toUri().scheme == "blob"
```

Comment:
I think I would have passed `isBlob` as a paramter to `toReponse`, because in `fetch` we already have the URL and can check if it's a blob or not (similar to our data URL check).